hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -400,6 +400,8 @@ public void deregisterAllListeners(@Nonnull String serviceName, @Nonnull String
         if (segment != null) {
             segment.removeRegistrations(topic);
         }
+        invokeOnStableClusterSerial(nodeEngine, new DeregistrationOperationSupplier(
```

Comment:
I'm not too familiar with this code, did you consult it with somebody from the core team? Looks like a pretty significant change to call an operation on all members where no operation was called before...

Reply:
The addition/removal of a single registration propagates across all members. Strangely, the removal of all registrations of a topic only applies to the local member, which causes leftover event registrations on other members. Is @hazelcast/core team happy with this change?

Reply:
`deregisterAllListeners` should only operate on the local registrations. This is unfortunately not documented in its javadoc.
This method is used on proxy destruction, which has its own means of propagating across all members of the cluster. Changing `deregisterAllListeners` to invoke operations on other members will cause duplication of deregistration operations on all members of the cluster or may even result in deadlock.
If you need a method to deregister all listeners from a topic cluster-wide, I would suggest you implement a new method in `EventService` (and maybe rename existing method to `deregisterAllLocalListeners` so it is clearer).